addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java

```diff
@@ -0,0 +1,430 @@
+	}
+
+	public synchronized void sendMessage(RFXComMessage msg) {
+
+		((RFXComBaseMessage) msg).seqNbr = getNextSeqNumber();
+		byte[] data = msg.decodeMessage();
+
+		logger.debug("Transmitting message '{}'", msg);
+		logger.trace("Transmitting data: {}",
+				DatatypeConverter.printHexBinary(data));
+
+		setResponseMessage(null);
+
+		try {
+			connector.sendMessage(data);
+		} catch (IOException e) {
+			logger.error("Message sending to RFXCOM controller failed.", e);
+		}
+
+		try {
+
+			RFXComTransmitterMessage resp = null;
+			synchronized (notifierObject) {
+				notifierObject.wait(timeout);
+				resp = getResponseMessage();
+			}
+
+			if (resp != null) {
+				switch (resp.response) {
+				case ACK:
+				case ACK_DELAYED:
+					logger.debug(
+							"Command succesfully transmitted, '{}' received",
+							resp.response);
+					break;
+
+				case NAK:
+				case NAK_INVALID_AC_ADDRESS:
+				case UNKNOWN:
+					logger.error("Command transmit failed, '{}' received",
+							resp.response);
+					break;
+				}
+			} else {
+				logger.warn("No response received from transceiver");
+			}
+
+		} catch (InterruptedException ie) {
+			logger.error(
+					"No acknowledge received from RFXCOM controller, timeout {}ms ",
+					timeout);
+		}
+	}
+
+	private class MessageLister implements RFXComEventListener {
+
+		@Override
+		public void packetReceived(EventObject event, byte[] packet) {
+
+			try {
+				RFXComMessage message = RFXComMessageFactory.getMessage(packet);
+				logger.debug("Message received: {}", message);
+
+				if (message instanceof RFXComInterfaceMessage) {
+					logger.debug("Received interface message: ", message);
+					RFXComInterfaceMessage msg = (RFXComInterfaceMessage) message;
+					logger.info(
+							"RFXCOM transceiver/receiver type: {}, hw version: {}.{}, fw version: {}",
+							msg.transceiverType, msg.hardwareVersion1,
+							msg.hardwareVersion2, msg.firmwareVersion);
+
+				} else if (message instanceof RFXComTransmitterMessage) {
+					RFXComTransmitterMessage resp = (RFXComTransmitterMessage) message;
+
+					byte seqNbr = getSeqNumber();
+					if (resp.seqNbr == seqNbr) {
+						logger.debug("Transmitter response received: {}",
+								message.toString());
+						setResponseMessage(resp);
+						synchronized (notifierObject) {
+							notifierObject.notify();
+						}
+					} else {
+						logger.warn(
+								"Sequnce number '{}' does not match, expecting number '{}'",
+								resp.seqNbr, seqNbr);
+					}
+
+				} else {
+
+					for (DeviceMessageListener deviceStatusListener : deviceStatusListeners) {
+						try {
+							deviceStatusListener.onDeviceMessageReceived(
+									getThing().getUID(), message);
+						} catch (Exception e) {
+							logger.error(
+									"An exception occurred while calling the DeviceStatusListener",
+									e);
+						}
+					}
+				}
+			} catch (RFXComException e) {
+				logger.error("Error occured during packet receiving, data: {}",
```

Comment:
instead of logging, better set the status to OFFLINE and provide the exception message as a status details